src/main/cpp/profiler.h

```diff
@@ -79,12 +90,18 @@ class Profiler {
 
     Processor *processor;
 
-    SignalHandler handler_;
+    SignalHandler* handler_;
+
+    bool reloadConfig;
```

Comment:
Are all the reads and writes of reloadConfig on the same thread? It's not 100% clear to me. It looks like they can get written from an arbitrary JNI method which maybe called from an arbitrary thread. In that case you need some coordination of the reading and writing of the bool in question.